File: sol/solution/0101-0200/0108/Solution.scala
/**
 * Definition for a binary tree node.
 * class TreeNode(_value: Int = 0, _left: TreeNode = null, _right: TreeNode = null) {
 *   var value: Int = _value
 *   var left: TreeNode = _left
 *   var right: TreeNode = _right
 * }
 */
object Solution {
    def sortedArrayToBST(nums: Array[Int]): TreeNode = {
        def buildBST(left: Int, right: Int): TreeNode = {
            if (left > right) null
            else {
                val mid = left + (right - left) / 2
                val root = new TreeNode(nums(mid))
                root.left = buildBST(left, mid - 1)
                root.right = buildBST(mid + 1, right)
                root
            }
        }
        buildBST(0, nums.length - 1)
    }
}

File: sol/solution/0101-0200/0108/Solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sortedArrayToBST(nums []int) *TreeNode {
    var buildBST func(left, right int) *TreeNode
    buildBST = func(left, right int) *TreeNode {
        if left > right {
            return nil
        }
        mid := left + (right - left) / 2
        root := &TreeNode{Val: nums[mid]}
        root.Left = buildBST(left, mid - 1)
        root.Right = buildBST(mid + 1, right)
        return root
    }
    return buildBST(0, len(nums) - 1)
}